docs/querying.md

```diff
@@ -2,10 +2,83 @@
 
 The query macro and where clause combine to give you full control over your query.
 
+## Where
+
+Where is using a QueryBuilder that allows you to chain where clauses together to build up a complete query.
+```crystal
+posts = Post.where(published: true, author_id: User.first.id)
+```
+
+It supports different operators:
+```crystal
+Post.where(:created_at, :gt, Time.now - 7.days)
+```
+
+Supported operators are :eq, :gteq, :lteq, :neq, :gt, :lt, :nlt, :ngt, :ltgt, :in, :nin, :like, :nlike
+
+
+This allows you create snippets of clauses that can be easily chained together similar to scopes in Rails.
+```crystal
+class Post < Granite::Base
+...
+  def live
+    where(published: true)
+  end
+
+  def draft
+    where(published: false)
+  end
+
+  def last_week
```

Comment:
In these scopes (not really related to this PR) is there way to declare them as not_nil similar to the `field!` macro?  Or do nil values have to be handled on the user's end?

Reply:
Interesting question!  I think the `where` would return `Array(Model)` not `Array(Model | Nil)` so there is no need for `!` in this case, right?

Reply:
Right, which in some cases could be empty.  But that would just be an empty `Array(Model)` so nil shouldnt be an issue i'd think.